backend/app/crud/credentials.py

```
@@ -107,6 +107,26 @@
     return None
 
 
+def get_full_provider_credential(
+    *, session: Session, org_id: int, provider: str, project_id: Optional[int] = None
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)